tgui/packages/tgui/interfaces/TrainingMachine.js

```diff
@@ -0,0 +1,74 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Box, Section, Knob, Flex, LabeledControls, Divider } from '../components';
+import { Window } from '../layouts';
+
+export const TrainingMachine = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [
+    speed,
+    setSpeed,
+  ] = useLocalState(context, 'speed', 1);
+  const [
+    range,
+    setRange,
+  ] = useLocalState(context, 'range', 1);
+  return (
+    <Window
+      width={230}
+      height={150}
+      title="AURUMILL">
+      <Window.Content>
+        <Section fill title="Training Machine">
+          <LabeledControls m={1}>
+            <LabeledControls.Item label="Speed">
+              <Knob
+                inline
+                size={1.2}
+                step={.5}
+                stepPixelSize={10}
+                value={speed}
+                minValue={1}
+                maxValue={10}
+                onDrag={(e, value) => {
+                  act("movespeed", { movespeed: value });
```

Comment:
why range is stored in local state, when it can be a part of ui_data?

Reply:
for future reference, turns out the issue was me forgetting to return TRUE in ui_act()